docs/advanced/featureflags.md

```diff
@@ -0,0 +1,280 @@
+# Feature Flags
+
+Feature Flags are a system which allow developers to gate a set of features behind some set of required flags, that being registered elements, gameplay mechanics, data pack entries or some other unique system to your mod.
+
+A common use case would be gating experimental features/elements behind a experimental flag, allowing users to easily switch them on and play around with them before they are finalized.
```

Comment:
what do you mean here exactly?
do you mean all the pale oak blocks are gated cause they depend on the pale oak wood type?
if so thats just simply not true, `WoodType` and its paired `BlockSetType` cannot be gated behind flags, its each individual `Block` thats defining the flag thats gating them, which would fall under `registered elements,`.

Reply:
No, I just wanted something like "You can also gate your own blocks behind vanilla feature flags. For example if you were to add to the set of Pale Oak wood blocks, you'd only want those to show up if Pale Oak is enabled." (but more general because using Pale Oak as an example is gonna be outdated by the next version)